test/api/pegjs-api.spec.js

```diff
@@ -271,6 +271,9 @@ describe("Peggy API", () => {
 
           it("rule name", () => check("RULE_1", source, "RULE_1", "peg$parseRULE_1() {"));
           it("labelled rule name", () => check("RULE_2 'named'", source, "RULE_2", "peg$parseRULE_2() {"));
+          it("literal expression", () => check("'a'", source, null, "input.charCodeAt(peg$currPos) === 97"));
+          it("chars expression", () => check("[abc]", source, null, "peg$r0.test(input.charAt(peg$currPos))"));
+          it("rule expression", () => check("RULE_2", source, null, "= peg$parseRULE_2();"));
```

Comment:
I thought I posted this yesterday, but I can't find my comment so...

`peg$parseRULE_2` would actually work right now, but relies on the generated code for the use of the rule coming after the generated code for the definition. We don't need the `=`, but I think the `();` should be left to differentiate it from the definition, `peg$parseRULE_2() {`, which you used above. It might even be a good idea to add a check that the generated chunk only occurs once. 

Also, I ended up using a complicated regex to test a choice expression - not sure if that falls into the "not nice" category, but I couldn't see a better way of doing it.

Reply:
Yes